base/sparse/sparsematrix.jl

```diff
@@ -1179,87 +1172,62 @@ function getindex_I_sorted_bsearch_I{Tv,Ti}(A::SparseMatrixCSC{Tv,Ti}, I::Abstra
     return SparseMatrixCSC(nI, nJ, colptrS, rowvalS, nzvalS)
 end
 
+function permute_rows!{Tv,Ti}(S::SparseMatrixCSC{Tv,Ti}, pI::Vector{Int})
+    (m, n) = size(S)
+    colptrS = S.colptr; rowvalS = S.rowval; nzvalS = S.nzval
+    # preallocate temporary sort space
+    nr = min(nnz(S), m)
+    rowperm = Array(Int, nr)
+    rowvalTemp = Array(Ti, nr)
+    nzvalTemp = Array(Tv, nr)
 
-function getindex_general{Tv,Ti}(A::SparseMatrixCSC{Tv,Ti}, I::AbstractVector, J::AbstractVector)
-    # Anything for indexing rows.
-    # This sorts I first then does some trick with constructing the transpose.
-    (m, n) = size(A)
-    nI = length(I)
-    nJ = length(J)
+    @inbounds for j in 1:n
+        rowrange = colptrS[j]:(colptrS[j+1]-1)
+        nr = length(rowrange)
+        (nr > 0) || continue
+        k = 1
+        for i in rowrange
+            rowA = rowvalS[i]
+            rowvalTemp[k] = pI[rowA]
+            nzvalTemp[k] = nzvalS[i]
+            k += 1
+        end
+        sortperm!(pointer_to_array(pointer(rowperm), nr), pointer_to_array(pointer(rowvalTemp), nr))
```

Comment:
`pointer_to_array` allocates small amounts of memory, and inside a large loop like the one here, I wonder if it creates a lot of garbage. I am curious if subarrays are fast enough now for such usage. @timholy ?